Tests/AsyncHTTPClientTests/HTTPClientTestUtils.swift

```diff
@@ -804,6 +813,10 @@ internal final class HTTPBinHandler: ChannelInboundHandler {
                 builder.add(buff)
                 self.resps.append(builder)
                 return
+            case "/sendheaderandwait":
+                // sends some headers and waits indefinitely afterwards
+                context.writeAndFlush(wrapOutboundOut(.head(HTTPResponseHead(version: HTTPVersion(major: 1, minor: 1), status: .ok))), promise: nil)
+                return
```

Comment:
Please create a small `ChannelInboundHandler` for this in the specific test case. There is so much stuff in this single handler and all is mixed.